src/merkle/merkle_tree.rs

```diff
@@ -57,8 +56,8 @@ impl MerkleTree {
     /// Returns the depth of this Merkle tree.
     ///
     /// Merkle tree of depth 1 has two leaves, depth 2 has four leaves etc.
-    pub fn depth(&self) -> u32 {
-        log2(self.nodes.len() / 2)
+    pub fn depth(&self) -> u8 {
+        (self.nodes.len() as u8 / 2).trailing_zeros() as u8
```

Comment:
I'd prefer to keep using `log2()` function here. It just does `trailing_zeros()` internally - but I think it makes the intent more clear.

Reply:
Done!